pkg/localstore/sampler.go

```diff
@@ -180,8 +188,28 @@ func (db *DB) ReserveSample(
 		} else {
 			currentMaxAddr = swarm.NewAddress(make([]byte, 32))
 		}
-		if le(item.Bytes(), currentMaxAddr.Bytes()) || len(sampleItems) < sampleSize {
-			insert(item)
+		if le(item.transformedAddress.Bytes(), currentMaxAddr.Bytes()) || len(sampleItems) < sampleSize {
+
+			chunk := swarm.NewChunk(swarm.NewAddress(item.chunkItem.Address), item.chunkItem.Data)
+
+			stamp := postage.NewStamp(item.chunkItem.BatchID, item.chunkItem.Index, item.chunkItem.Timestamp, item.chunkItem.Sig)
+
+			stampData, err := stamp.MarshalBinary()
+			if err != nil {
+				logger.Info("error marshaling stamp for chunk", chunk.Address(), err)
+				continue
+			}
+			_, err = db.validStamp(chunk, stampData)
+			if err == nil {
+				if !soc.Valid(chunk) && !cac.Valid(chunk) {
```

Comment:
Since most chunks are are cacs, checking `cac.Valid(..)` first in the if statement is a simple optimization to have here.